superset/assets/src/dashboard/components/filterscope/FilterScopeSelector.jsx

```diff
@@ -0,0 +1,518 @@
+  setDirectPathToChild: PropTypes.func.isRequired,
+  onCloseModal: PropTypes.func.isRequired,
+};
+
+export default class FilterScopeSelector extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    const {
+      dashboardFilters,
+      filterImmuneSlices,
+      filterImmuneSliceFields,
+      layout,
+    } = props;
+
+    if (Object.keys(dashboardFilters).length > 0) {
+      // display filter fields in tree structure
+      const filterFieldNodes = getFilterFieldNodesTree({
+        dashboardFilters,
+      });
+      // filterFieldNodes root node is dashboard_root component,
+      // so that we can offer a select/deselect all link
+      const filtersNodes = filterFieldNodes[0].children;
+      this.allfilterFields = [];
+      filtersNodes.forEach(({ children }) => {
+        children.forEach(child => {
+          this.allfilterFields.push(child.value);
+        });
+      });
+      this.defaultFilterKey = filtersNodes[0].children[0].value;
+
+      // build FilterScopeTree object for each filterKey
+      const filterScopeMap = Object.values(dashboardFilters).reduce(
+        (map, { chartId: filterId, columns }) => {
+          const filterScopeByChartId = Object.keys(columns).reduce(
+            (mapByChartId, columnName) => {
+              const filterKey = getDashboardFilterKey({
+                chartId: filterId,
+                column: columnName,
+              });
+              const nodes = getFilterScopeNodesTree({
+                components: layout,
+                filterFields: [filterKey],
+                selectedChartId: filterId,
+              });
+              const checked = getCurrentScopeChartIds({
+                scopeComponentIds: ['ROOT_ID'], // dashboardFilters[chartId].scopes[columnName],
+                filterField: columnName,
+                filterImmuneSlices,
+                filterImmuneSliceFields,
+                components: layout,
+              });
+              const expanded = getFilterScopeParentNodes(nodes, 1);
+              return {
+                ...mapByChartId,
+                [filterKey]: {
+                  // unfiltered nodes
+                  nodes,
+                  // filtered nodes in display if searchText is not empty
+                  nodesFiltered: [...nodes],
+                  checked,
+                  expanded,
+                },
+              };
+            },
+            {},
+          );
+
+          return {
+            ...map,
+            ...filterScopeByChartId,
+          };
+        },
+        {},
+      );
+
+      // initial state: active defaultFilerKey
+      const { chartId } = getChartIdAndColumnFromFilterKey(
+        this.defaultFilterKey,
+      );
+      const checkedFilterFields = [];
+      const activeFilterField = this.defaultFilterKey;
+      // expand defaultFilterKey in filter field tree
+      const expandedFilterIds = [ALL_FILTERS].concat(chartId);
```

Comment:
is chartId not included in `ALL_FILTERS` seems like it wouldn't be all the filters then

Reply:
`ALL_FILTERS` is a node value. just like dashboard's root's value is `ROOT_ID`. 
`expandedFilterIds` expects an array of node values that should be displayed as expanded tree branch. I think `ALL_FILTERS` seems misleading here, so i rename it to be `ALL_FILTERS_ROOT`.